src/inet/TCPEndPointImplSockets.cpp

```diff
@@ -734,6 +736,10 @@ CHIP_ERROR TCPEndPointImplSockets::GetSocket(IPAddressType addrType)
         {
             return CHIP_ERROR_POSIX(errno);
         }
+        auto connectionCleanup = ScopeExit([&]() {
```

Comment:
@bzbarsky-apple pretty sure this was the cause of the `SEGV on unknown address ` you saw; basically if the port is already in use, `StartWatchingSocket` below fails & it all falls apart; it would be nice if our test could test for this generically by simply reserving the port before starting the test, which theoretically would cause this failure; I'm unsure how to do that